nixos/modules/config/i18n.nix

```diff
@@ -118,6 +118,25 @@
   ###### implementation
 
   config = {
+    warnings =
+      lib.optional
+        (
+          (lib.subtractLists config.i18n.supportedLocales (
+            builtins.map
+              (l: (lib.replaceStrings [ "utf8" "utf-8" "UTF8" ] [ "UTF-8" "UTF-8" "UTF-8" ] l) + "/UTF-8")
+              (
+                config.i18n.defaultLocale
+                ++ config.i18n.extraLocales
+                ++ (lib.attrValues (lib.filterAttrs (n: v: n != "LANGUAGE") config.i18n.extraLocaleSettings))
+              )
+          )) != [ ]
+        )
+        ''
+          It looks like that you've overriden `i18n.supportedLocales`, but
+          forgot to include some locales specified in `i18n.defaultLocale`,
+          `i18n.extraLocales` or `i18n.extraLocaleSettings`. If you're trying to
+          install additional locales, use `i18n.extraLocales` instead.
+        '';
```

Comment:
Hmm I know I am sort of asking for perhaps too much. Would it be possible to generate a slightly more detailed warning message? Maybe tell the user _which_ setting (`i18n.defaultLocale`, `i18n.extraLocales` or `i18n.extraLocaleSettings`) includes which locale that is missing from `supportedLocales`? TBH I'm not sure exactly how to implement this, but I am under the impression you are pretty competent and can figure out how to do that :).

Reply:
IMO this is more complexity than it's worth. The warning is meant to gently nudge the user to a better option in case they were mis-using it, rather than to serve as a versatile diagnostic tool. If you're getting this warning, the likely best course of action for you is to migrate to `extraLocales`.

But let me know what you think.